services/minespace-web/src/components/dashboard/mine/reports/Reports.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { FC, useEffect, useState } from "react";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import PlusCircleFilled from "@ant-design/icons/PlusCircleFilled";
+import { Button, Col, Row, Typography } from "antd";
+import moment from "moment";
+import PropTypes from "prop-types";
+import {
+  createMineReport,
+  fetchMineReports,
+  updateMineReport,
+} from "@mds/common/redux/actionCreators/reportActionCreator";
+import { closeModal, openModal } from "@mds/common/redux/actions/modalActions";
+import { getMineReports } from "@mds/common/redux/selectors/reportSelectors";
+import { getMineReportDefinitionOptions } from "@mds/common/redux/reducers/staticContentReducer";
+import CustomPropTypes from "@/customPropTypes";
+import ReportsTable from "@/components/dashboard/mine/reports/ReportsTable";
+import { modalConfig } from "@/components/modalContent/config";
+import AuthorizationWrapper from "@/components/common/wrappers/AuthorizationWrapper";
+import { IMine, IMineReport } from "@mds/common";
+import { ActionCreator } from "@mds/common/interfaces/actionCreator";
+
+interface ReportsProps {
+  mine: IMine;
+  mineReports: IMineReport[];
+  mineReportDefinitionOptions: any[];
+  updateMineReport: ActionCreator<typeof updateMineReport>;
+  createMineReport: ActionCreator<typeof createMineReport>;
+  fetchMineReports: ActionCreator<typeof fetchMineReports>;
+  openModal: (any?) => void;
+  closeModal: (any?) => void;
+}
+
+export const Reports: FC<ReportsProps> = ({
+  mine,
+  mineReports,
+  mineReportDefinitionOptions,
+  ...props
+}) => {
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [report, setReport] = useState(null);
+
+  useEffect(() => {
+    props.fetchMineReports(mine.mine_guid).then(() => {
+      setIsLoaded(true);
```

Comment:
would recommend the isMounted trick (or if you find something similar) to avoid the "can't do state update on unmounted component" error when setIsLoaded gets called.

Reply:
Oooh, good opportunity to switch to those redux hooks.  On it.